### PR TITLE
feat: describe LNodeType

### LINKED ISSUE #9: describeLNodeType
As a user of `scl-diff` I would like to have the SCL element `LNodeType` described.

**Requirements**
- return a `LNodeTypeDescription`
```ts
interface DODescription extends NamingDescription {
  /** Type attribute accessControl */
  accessControl: string;
  /** Type attribute transient defaulting to false */
  transient: boolean;
  /** Type attribute referencing a DOType */
  type: DOTypeDescription;
}

export interface LNodeTypeDescription extends NamingDescription {
  /** Required attribute lnClass */
  lnClass: string;
  /** Optional attribute iedType*/
  iedType?: string;
  /** Child elements SDO key is required name attribute of DA child */
  dos: Record<string, DODescription>;
}
```

### PR DESCRIPTION
**Pull request by JakobVogelsang**
Closes #9 

Finally last element in the `DataTypeTemplate` section. I wonder what nice things we can do with that. Maybe a discussion with a collection would be beneficial.

### CODE REVIEW

**Comment by danyill** [comment]
Ooh! How exciting, here are some initial thoughts:

* All "types" in the DTT section if they have the same hash, ids can be updated to just one. So datatype cleanup in the Cleanup editor can be vastly more efficient
* When we import IEDs we can hash their datatypes and if the hash exists we don't need to transfer the template across. This is more robust than we do at the moment and allows us to avoid duplicated datatypes based on `isEqualNode`
* When creating a type for some reason we can see if a hash of it already exists and if it does, we reference that one.
* Being able to check point a document and then being able to generate a "fast diff" to the check point where if there is a difference between the nodes we generate a text diff so you can see exactly what has changed. Could start with DTTs and build sections up slowly and hide behind a feature flag (with oscd-tree-grid).

What would be neat is to hash the DTTs and have them accessible as a service (dare I say it a service in "core"! 😜)

**Comment by ca-d** [comment]
> * All "types" in the DTT section if they have the same hash, ids can be updated to just one.

While this is a perfectly fine idea for the use case of bottom-up engineering, for top-down engineering I can see a real use case for separate, differently named data types which happen to have exactly the same shape. By distinguishing different uses for identically shaped types (e.g. Mod and Beh), we keep the ability to later update the one without updating the other. If we just unify all the types, making a project-wide change to the data types later may involve the hassle of going through all functions and LNodes and seeing where the type is being used for what purpose.

This would not present an issue (for top-down engineering I could simply refrain from unifying my data type templates), I just wanted to point that out so we don't bake template de-duping into our tools in some fundamental and non-configurable way.

File: utils.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+      return sortedRecord;
+    }, {});
+}
+
+/** @returns DOType/DAType/EnumType reference by the elements type attribute */
+export function referencedDataType(element: Element): Element | undefined {
+  return Array.from(element.closest("DataTypeTemplates")?.children ?? []).find(
+    (sibling) => sibling.getAttribute("id") === element.getAttribute("type")
+  )!;
```
Comment (ca-d) [inline utils.ts]: A small performance improvement could be got by the not much less readable

```suggestion
  const elementType = element.getAttribute("type");
  return Array.from(element.closest("DataTypeTemplates")?.children ?? []).find(
    (sibling) => sibling.getAttribute("id") === elementType
  )!;
```

**Review by ca-d — APPROVED** [review]
LGTM! I'm skipping my ceterum censeo about test naming; your scheme is fine, too :wink:

File: utils.ts (hunk 1 above)
Comment (JakobVogelsang) [reply utils.ts]: True and better readable. Thank you

**Pull request merged**